lib/order.rb

```diff
@@ -0,0 +1,28 @@
+class Order
+
+  attr_reader :closed, :basket
+
+  def initialize(menu)
+    @basket = []
+    @menu = menu
```

Comment:
Yep, they could be totally decoupled, but what would be the right-to-exist for the order class if it cannot even check some basics in the menu. But yeah, I could literally just check if a dish is in the menu in the restaurant class and also would need give the order the price for a dish when adding it. Right now, I just let the Order look up the prices in the menu itself.

Reply:
To me the Restaurant seems like it would be coordinating between the two things. The Restaurant can be aware of the menu and of the order and then add items and price to the order. Order could even be generic to the point where it can take things other than food, it just need to be able to take an item with a price and then add it to a list of items with prices and then return a total cost when asked. Does that make sense? If not Im happy to jump on a call!